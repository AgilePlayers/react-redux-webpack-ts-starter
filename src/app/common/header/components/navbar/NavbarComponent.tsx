import React from 'react';
import { Link } from 'react-router-dom';
import './navbar.scss';

const NavbarComponent: React.SFC = () => (
  <ul className="navbar">
    <li className="navbar-item">
      <Link to="/" className="navbar-item-link">
        Home
      </Link>
    </li>
    <li className="navbar-item">
      <Link to="/books" className="navbar-item-link">
        Books
      </Link>
    </li>
  </ul>
);

export default NavbarComponent;
