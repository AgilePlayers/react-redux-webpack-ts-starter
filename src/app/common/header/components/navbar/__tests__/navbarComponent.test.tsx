import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import NavbarComponent from '../NavbarComponent';

describe('common / navbar / NavbarComponent', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<NavbarComponent />);
  });

  test('should render', () => {
    expect(component.length).toEqual(1);
  });

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('should contains elements with "navbar" and "navbar-item" class', () => {
    expect(component.find('ul.navbar').length).toEqual(1);
    expect(component.find('li.navbar-item').length).toEqual(2);
  });
});
