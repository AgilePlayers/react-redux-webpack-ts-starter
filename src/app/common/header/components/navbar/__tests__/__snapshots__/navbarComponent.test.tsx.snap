// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common / navbar / NavbarComponent should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavbarComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="navbar-item"
        >
          <Link
            className="navbar-item-link"
            replace={false}
            to="/"
          >
            Home
          </Link>
        </li>,
        <li
          className="navbar-item"
        >
          <Link
            className="navbar-item-link"
            replace={false}
            to="/books"
          >
            Books
          </Link>
        </li>,
      ],
      "className": "navbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="navbar-item-link"
            replace={false}
            to="/"
          >
            Home
          </Link>,
          "className": "navbar-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Home",
            "className": "navbar-item-link",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="navbar-item-link"
            replace={false}
            to="/books"
          >
            Books
          </Link>,
          "className": "navbar-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Books",
            "className": "navbar-item-link",
            "replace": false,
            "to": "/books",
          },
          "ref": null,
          "rendered": "Books",
          "type": [Function],
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="navbar-item"
          >
            <Link
              className="navbar-item-link"
              replace={false}
              to="/"
            >
              Home
            </Link>
          </li>,
          <li
            className="navbar-item"
          >
            <Link
              className="navbar-item-link"
              replace={false}
              to="/books"
            >
              Books
            </Link>
          </li>,
        ],
        "className": "navbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="navbar-item-link"
              replace={false}
              to="/"
            >
              Home
            </Link>,
            "className": "navbar-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Home",
              "className": "navbar-item-link",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="navbar-item-link"
              replace={false}
              to="/books"
            >
              Books
            </Link>,
            "className": "navbar-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Books",
              "className": "navbar-item-link",
              "replace": false,
              "to": "/books",
            },
            "ref": null,
            "rendered": "Books",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
<<<<<<< HEAD:src/app/common/header/components/navbar/__tests__/__snapshots__/navbarComponent.test.tsx.snap
          "getSnapshotBeforeUpdate": true,
=======
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
>>>>>>> npm-update:src-ts/app/common/header/components/navbar/__tests__/__snapshots__/navbarComponent.test.tsx.snap
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
